### ISSUE TITLE
Bug in beta branch - Toggleing Bulbs of MiLight Hub is not working

### ISSUE THREAD

**Issue by hdhdu** [diyhue/diyHue#689]
**Describe the bug**

Installed Beta branch.
Thank you for that great UI-Work!

I have added all Lights that are registered in my milight hub (https://github.com/sidoh/esp8266_milight_hub). Those lights works perfectly with master-branch-version.
With the beta-branch-version nothing happend, when i try to switch light on or off.

**Steps to Reproduce**

Added light of milight hub to diyhue.
-> Click on the bulb of the added light (alert)-> nothing happend.
-> Try to toggle the main toggle in the right top corner -> nothing happens
Added Light in Hue APP (v4) and add it to a room.
-> Try to switch light on/off ->nothing happend

**Expected behavior**

I would expect, that I can toggle bulbs of milight hub. 

I have tried to add the bulb with a device ID with 0x and without 0x.

There are no errors in the log.

The only thing I saw are a lot of requests to "/api/invalid/lights". But I don't know if those are regular requests or not.
**Logs**

192.168.178.65 - - [30/Aug/2021 12:09:21] "GET /api/invalid/lights HTTP/1.1" 200 -
192.168.178.65 - - [30/Aug/2021 12:09:21] "GET /api/cfcf916a4eab2b122d829dbfdf75b6d1/lights HTTP/1.1" 200 -
192.168.178.65 - - [30/Aug/2021 12:09:21] "GET /api/invalid/lights HTTP/1.1" 200 -
192.168.178.65 - - [30/Aug/2021 12:09:21] "GET /api/cfcf916a4eab2b122d829dbfdf75b6d1/lights HTTP/1.1" 200 -
192.168.178.65 - - [30/Aug/2021 12:09:22] "GET /api/invalid/lights HTTP/1.1" 200 -
192.168.178.65 - - [30/Aug/2021 12:09:22] "GET /api/cfcf916a4eab2b122d829dbfdf75b6d1/lights HTTP/1.1" 200 -
192.168.178.88 - - [30/Aug/2021 12:09:22] "GET /lights HTTP/1.1" 200 -
192.168.178.65 - - [30/Aug/2021 12:09:23] "GET /api/invalid/lights HTTP/1.1" 200 -
192.168.178.65 - - [30/Aug/2021 12:09:23] "GET /api/cfcf916a4eab2b122d829dbfdf75b6d1/lights HTTP/1.1" 200 -
192.168.178.65 - - [30/Aug/2021 12:09:23] "GET /api/invalid/lights HTTP/1.1" 200 -
192.168.178.65 - - [30/Aug/2021 12:09:23] "GET /api/cfcf916a4eab2b122d829dbfdf75b6d1/lights HTTP/1.1" 200 -
192.168.178.65 - - [30/Aug/2021 12:09:24] "GET /api/invalid/lights HTTP/1.1" 200 -

**Docker Info (please complete the following information):**
No Docker. Installed Version by checking out beta branch and running install.sh

**Checklist**
- [x] I confirm that this is a bug report
- [x] I confirm that I have completed this template to the best of my ability

**Additional context**

**Comment by hdhdu** [comment]
Sorry, I forgot to tell you the IPs of my log:
192.168.178.65 is the ip of my raspberry PI with running diyhue.
192.168.178.88 is my notebook with running diyhue interface.

My milight hub is running on 192.168.178.119. There are no logs with that IP when I try to toggle a light of milight hub.

**Comment by mariusmotea** [comment]
Hi.

Please provide me debug logs during the lights switch. The provided log has no value since contain basic http requests.

**Comment by hdhdu** [comment]
Hey, 
these are the entries when toggeling in hue app: 
`192.168.178.65 - - [30/Aug/2021 13:05:17] "GET /api/invalid/lights HTTP/1.1" 200 -
192.168.178.65 - - [30/Aug/2021 13:05:17] "GET /api/cfcf916a4eab2b122d829dbfdf75b6d1/lights HTTP/1.1" 200 -
2021-08-30 13:05:17,414 - flaskUI.v2restapi - DEBUG - {'on': {'on': False}}
2021-08-30 13:05:17,417 - flaskUI.v2restapi - DEBUG - Cache Miss light
192.168.178.24 - - [30/Aug/2021 13:05:17] "PUT /clip/v2/resource/light/e336c0ad-716b-4012-aa59-e32e37868d24 HTTP/1.1" 200 -
192.168.178.65 - - [30/Aug/2021 13:05:17] "GET /api/invalid/lights HTTP/1.1" 200 -
`

these are the loggs when toggle one created room with bulbs in the diyhue browser view:
`192.168.178.88 - - [30/Aug/2021 13:07:24] "GET /api/b1c2662c097a11ec8ab3b827eb924b05 HTTP/1.1" 200 -
192.168.178.65 - - [30/Aug/2021 13:07:25] "GET /api/invalid/lights HTTP/1.1" 200 -
192.168.178.65 - - [30/Aug/2021 13:07:25] "GET /api/cfcf916a4eab2b122d829dbfdf75b6d1/lights HTTP/1.1" 200 -
{'on': False}
2021-08-30 13:07:25,451 - functions.rules - DEBUG - Processing rules for Wohnzimmer
[{'success': {'/groups/1/action/on': False}}]
192.168.178.88 - - [30/Aug/2021 13:07:25] "PUT /api/b1c2662c097a11ec8ab3b827eb924b05/groups/1/action HTTP/1.1" 200 -
192.168.178.65 - - [30/Aug/2021 13:07:25] "GET /api/invalid/lights HTTP/1.1" 200 -
`

And this ist the log when clicking on the bulb in the lights section ("alerting") of diyhue :
`192.168.178.65 - - [30/Aug/2021 13:09:23] "GET /api/invalid/lights HTTP/1.1" 200 -
192.168.178.65 - - [30/Aug/2021 13:09:23] "GET /api/cfcf916a4eab2b122d829dbfdf75b6d1/lights HTTP/1.1" 200 -
{'alert': 'select'}
[{'success': {'/lights/16/state/alert': 'select'}}]
192.168.178.88 - - [30/Aug/2021 13:09:23] "PUT /api/b1c2662c097a11ec8ab3b827eb924b05/lights/16/state HTTP/1.1" 200 -
192.168.178.65 - - [30/Aug/2021 13:09:23] "GET /api/invalid/lights HTTP/1.1" 200 -`

As you can see, there are no requests listet to the IP of my milight hub. 

This is how an created bulb looks like:
![image](https://user-images.githubusercontent.com/43926658/131336884-bbe27f1a-2b58-4a39-8c95-965b7ade1899.png)

And this is a screenshot of a filled out light form:
![image](https://user-images.githubusercontent.com/43926658/131337195-baa55e50-31ac-4652-b2a4-56bab7e2f901.png)

And the Bulb in milight hub:
![image](https://user-images.githubusercontent.com/43926658/131337219-9d282c85-689f-4e86-82f3-482df8586fff.png)

**Comment by mariusmotea** [comment]
No exception in the logs. In lights.yaml file all details where correctly saved?

**Comment by hdhdu** [comment]
In my opinion, lights.yaml is looking good, but I don't know how it should look like.
I have added the lights.yaml. (I had to rename it to txt that I can attach it) 
[lights.txt](https://github.com/diyhue/diyHue/files/7076744/lights.txt)

**Comment by mariusmotea** [comment]
Is ok. I will debug this issue next week once i return from vacation. Unfortunately I had no clue currently wat is the issue.

**Comment by hdhdu** [comment]
Great news!
I have fixed it!

I have started a local http-server as a fake-milight-hub.
I have created a bulb with the IP of this server to have a look, if there are any requests to debug on the hub side. 
Actually the server did not receive any requests. AND when I killed the test-server, there are no errors in the diyhue-logs. I would have expected something like "host not reachable" or connection timeout". 
I have testet all possible selections in the dropdown and created a bulb for every provided protocol. 
The only request the testserver received was on "Autodetect". 
So I thought there is a problem with sending the request. And I started to debug your code. 
First of all, the milight module was missing in the
`/lights/protocols/__init__.py`
I have added it and then realized, that the code of 
`lights/protocols/milight.py` was full of not working lines of code. 

So I have run through my bulbs and tried to fixed it. 
It is working now for my RGBW and RGB+CCT (LCT001) bulbs.
I have attached the changed files. 
[milight.py.txt](https://github.com/diyhue/diyHue/files/7078173/milight.py.txt)
[__init__.py.txt](https://github.com/diyhue/diyHue/files/7078174/__init__.py.txt)
 I did not get to work, that the hue app only shows rgb color sitch without the white (warm white to cold white) switch. Is there a bulb type which makes the hue app only show the color-switch for the bulb?

Thank you again for that great project! It was fun fixing this problem and debug through the code. 
Happy vacation!
